File: 027_vector/main.py
# -*- coding: utf-8 -*-

def print_vector(v):
    print(v)


def main():
    print('initialize vector')
    v = [1, 2, 3]
    print_vector(v)
    print('')

    print('get length');
    print('length is {0}'.format(len(v)));
    print('');

    print('push');
    v.append(4);
    print_vector(v);
    print('');

    print('pop');
    v.pop();
    print_vector(v);
    print('');

    print('accessor []');
    print('[0] is {0}'.format(v[0]));
    print('');

    print('range for');
    for x in v:
        print('{0}'.format(x));
    print('');

if __name__ == '__main__':
    main()

File: 027_vector/main.rs
fn main() {
    println!("initialize vector");
    let mut v: Vec<i32> = vec![1, 2, 3];
    println!("{:?}", v);
    println!("");

    println!("get length");
    println!("length is {}", v.len());
    println!("");

    println!("push");
    v.push(4);
    println!("{:?}", v);
    println!("");

    println!("pop");
    v.pop();
    println!("{:?}", v);
    println!("");

    println!("accessor []");
    println!("[0] is {}", v[0]);
    println!("");

    println!("range for");
    for x in v {
        println!("{}", x);
    }
    println!("");
}

